charabia/src/segmenter/camel_case.rs

```diff
@@ -66,6 +47,31 @@ impl<'t> Iterator for CamelCaseParts<'t> {
     }
 }
 
+fn find_camel_case_boundary(str: &str) -> Option<usize> {
+    // CamelCase boundary consists of at least 2 code-points. Avoid iterating over shorter strings.
+    // Note that using `remainder.chars().count() == 1` may catch more cases (non-ASCII strings)
+    // but the main focus here is on " ", "-" and similar that are abundantly produced
+    // by `split_word_bounds()` in the Latin segmenter and mere `len()` performs better at that.
+    if str.len() < 2 {
+        return None;
+    }
+
```

Comment:
Omitting this optimization may now have negligible performance impact - hard to say without benchmark, but likely worth trying.

Reply:
It's a good question,
I found a random study [trying to predict the frequency of words depending on their size](https://math.wvu.edu/~hdiamond/Math222F17/Sigurd_et_al-2004-Studia_Linguistica.pdf),
It seems that, in English, there is a bit more than 3% chance of encountering a word containing less than 2 characters. 🤔
I agree about the fact that this optimization seems superfluous. ☺️

Reply:
Good point. I was hesitant to keep this in too, so I actually did the benchmark and saw an improvement of about 1-2%. That's very close to criterion's noise threshold but I was getting consistent numbers over multiple runs so I decided to keep it. Note that the numbers refer to performance of the whole latin segmenter. The impact on `split_camel_case_bounds()` alone would be higher but that's not of our interest here.

To see why optimizing for such a 'cornery' case yields any measurable improvement, one must look at the context where `split_camel_case_bounds()` [is used](https://github.com/meilisearch/charabia/blob/16665a6ced92d90bb0d5c5762ca7d23349f18113/charabia/src/segmenter/latin.rs#L20-L27):
```rust
fn segment_str<'o>(&self, s: &'o str) -> Box<dyn Iterator<Item = &'o str> + 'o> {
    let lemmas = s
        .split_word_bounds()
        .flat_map(|lemma| lemma.split_inclusive('\''))
        .flat_map(|lemma| lemma.split_camel_case_bounds());

    Box::new(lemmas)
}
```

Here, `s` is the whole string we're segmenting. We first break it on word boundaries, then on apostrophes and then finally on camelCase boundaries. The important detail is that `split_word_bounds()` breaks apart the spacing between words. For instance, `"foo bar".split_word_bounds()` gives `["foo", " ", "bar"]. This optimization targets exactly these spaces and other one-character word separators. Intuitively, in longer texts (like used in the [benchmarks](https://github.com/meilisearch/charabia/blob/16665a6ced92d90bb0d5c5762ca7d23349f18113/charabia/benches/bench.rs#L11-L12)) the frequency of 1-char strings seen by `split_camel_case_bounds()` is close to 50%.*

Though, I have no strong feeling about this, so if you think that the tiny performance boost is not worth it I'll happily remove this.

---

* This raises the question: is there any performance to gain by shuffling the order in which we split on the various breakpoints. I just wanted to keep the scope small.

Reply:
Well, I may know why you are seeing improvement actually, I completely forgot the separators, if we take separators into account, we have much more lemmas containing 1 character than the 3% I could expect.
mmmmh... 🤔 honestly I don't have any strong point of view on this.
In fact, it's better to skip the process on separators, but, in another hand 1-2% gain is not so much compared to other optimization that could be done elsewhere in the Library.